pkg/util/strategicpatch/patch.go

```diff
@@ -110,6 +110,28 @@ func RequireKeyUnchanged(key string) PreconditionFunc {
 	}
 }
 
+// RequireMetadataKeyUnchanged creates a precondition function that fails
+// if the metadata.key is present in the patch (indicating its value
+// has changed).
+func RequireMetadataKeyUnchanged(key string) PreconditionFunc {
```

Comment:
This precondition seems wrong.  Annotations and Labels are in metadata and I should be allowed to change them.